ranndy/models/vae.py

```diff
@@ -0,0 +1,76 @@
+# Based on: https://arxiv.org/pdf/1511.06349.pdf
+# TODO: Word dropout and historyless decoding
+
+from models.auto_encoder import SentenceAutoEncoder
+import tensorflow as tf
+
+
+class SentenceVAE(SentenceAutoEncoder):
+
+    def __init__(self, data_iterator, mode=tf.estimator.ModeKeys.TRAIN):
+
+        # Instance variables associated with VAE
+        self.encoding_mean = None
+        self.encoding_log_var = None # Note: Log(variance) is learned to ensure positivity
+        self.sampled_state = None
+        self.reconstruction_loss = None
+        self.kl_loss = None
+        self.kl_weight_scaling = 10 # TODO: Tune these values
+        self.kl_weight_offset = 5000
+
+        super().__init__(data_iterator, mode)
+
+    def _build_encoder(self):
+        self.encoder_cell = tf.nn.rnn_cell.LSTMCell(self.lstm_size, name="vae_encoder_cell")
+        output, encoder_state_tuple = tf.nn.dynamic_rnn(self.encoder_cell, self.enc_embedding_output,
```

Comment:
Not consistent with regular auto autoencoder where you used _ instead of output

Reply:
You're right, the output is unused. Fixed.